game/scripts/vscripts/Sections/Chapter4/SectionCommunication.ts

```diff
@@ -0,0 +1,145 @@
+                const kunkka = getOrError(context[kunkkaName] as CDOTA_BaseNPC | undefined);
+                kunkka.AddItemByName("item_lotus_orb");
+                kunkka.AddItemByName("item_invis_sword");
+                kunkka.AddItemByName("item_armlet");
+                kunkka.AddItemByName("item_phase_boots");
+                kunkka.AddItemByName("item_heart");
+                kunkka.AddItemByName("item_monkey_king_bar");
+            }),
+
+            tg.setCameraTarget(contex => contex[kunkkaName]),
+
+            tg.fork([
+                tg.moveUnit(context => context[kunkkaName], Vector(-1500, 4000, 256).__sub(Vector(300, 200))),
+                tg.moveUnit(context => context[lunaName], Vector(-1500, 4000, 256).__sub(Vector(500, 400))),
+            ]),
+
+            tg.immediate(context => {
+                const kunkka = getOrError(context[kunkkaName] as CDOTA_BaseNPC | undefined);
+                kunkka.FaceTowards(playerHero.GetAbsOrigin());
+                const luna = getOrError(context[lunaName] as CDOTA_BaseNPC | undefined);
+                luna.FaceTowards(playerHero.GetAbsOrigin());
+            }),
+
+            tg.setCameraTarget(playerHero),
+
+            tg.immediate(_ => goalPressVoiceChatButton.start()),
+            tg.waitForCommand(37), // Team voice
+            tg.immediate(_ => goalPressVoiceChatButton.complete()),
+
+            tg.immediate(_ => goalChatWheelWP.start()),
+            tg.waitForChatWheel(),
+            tg.immediate(_ => goalChatWheelWP.complete()),
+            tg.wait(2),
+
+            // Kunkka destroy items
+            tg.immediate(_ => freezePlayerHero(true)),
+            tg.setCameraTarget(context => context[kunkkaName]),
+            tg.loop(context => {
+                const kunkka = getOrError(context[kunkkaName] as CDOTA_BaseNPC | undefined);
+                context[kunkkaName] = kunkka;
+                for (let i = 0; i < 6; i++) {
+                    const item = kunkka.GetItemInSlot(i)
+                    if (item) {
+                        context["ItemToDestroy"] = item;
+                        return true;
+                    }
+                }
+                return false;
+
+            }, tg.seq([
+                tg.immediate(context => {
+                    context[kunkkaName].DropItemAtPosition(context[kunkkaName].GetAbsOrigin().__add(Vector(50, 50)), context["ItemToDestroy"]);
+                }),
+                tg.completeOnCheck(context => !context[kunkkaName].HasItemInInventory((context["ItemToDestroy"] as CDOTA_Item).GetName()), 0.8),
+                tg.immediate(context => {
+                    ExecuteOrderFromTable({
+                        OrderType: UnitOrder.ATTACK_TARGET,
+                        UnitIndex: context[kunkkaName].entindex(),
+                        TargetIndex: (context["ItemToDestroy"] as CDOTA_Item).GetContainer()!.entindex(),
+                    });
+                }),
+                tg.wait(1),
```

Comment:
Agreed. Maybe replace the 1 second wait with a completeOnCheck() on the container being destroyed? We wait until Kunkka does it, as soon as it can, so we don't spend time time just watching him do nothing.